src/components/App/App.js

```diff
@@ -247,24 +265,16 @@ function App() {
 
   const handleRegisterSubmit = (data) => {
     setIsLoading(true);
-    return auth
-      .register(data)
-      .then((res) => {
-        handleLogInSubmit(data);
-        history.push("/");
-        handleCloseModal();
-      })
-      .catch((error) => {
-        setIsLoading(false);
-        if (error.response.status === 409) {
-          showTinyPopup("User is already registered, please sign in");
-        } else if (error.response.status === 500) {
-          showTinyPopup("Registration failed, please try again");
-        } else {
-          showTinyPopup("error occured on our end, please try again later");
-        }
-      });
+    const registerRequest = () => {
+      return auth.register(data)
+        .then((res) => {
+          setLoginFailed(false);
+          return handleLogInSubmit(data);
+        });
+    };
+    handleSubmit(registerRequest);
```

Comment:
Good job that you've implemented `handleSubmit`

Reply:
i was actually on the process of implementing this. experienced some hiccups in refactoring my registerSubmit and started to question if what i was doing was correct, so i just decided to turn it in. THANK YOU SOO MUCH! :)